app/views/contexts/context.py

```diff
@@ -39,12 +43,28 @@ def __init__(
             self._response_metadata,
         )
 
+        self._path_finder = PathFinder(
+            self._schema,
+            self._answer_store,
+            self._list_store,
+            self._progress_store,
+            self._metadata,
+            self._response_metadata,
+        )
+
+        if self._location:
+            self._routing_path = self._path_finder.routing_path(
```

Comment:
Since you got router here already, you don't need path finder as well, it can just be `self._router.routing_path()`